testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/SseResource.java

```diff
@@ -157,6 +157,33 @@ public void run()
          }
       }.start();
    }
+   
+   
+   @GET
+   @Path("/events")
+   @Produces(MediaType.SERVER_SENT_EVENTS)
+   public void eventStream(@Context SseEventSink sink) throws IOException {
+       if (sink == null) {
+           throw new IllegalStateException("No client connected.");
+       }
+       this.eventSink = sink;
+       new Thread() {
+           public void run() {
+               for (int i = 0; i < 1000; i++) {
+                   if (!eventSink.isClosed()) {
+                       eventSink.send(sse.newEvent(Integer.toString(i)));
+                   } else {
+                       break;
+                   }
+               }
+           }
+       }.start();
+   }
+   @GET
+   @Path("/isopen")
+   public boolean isOpen() {
+       return this.eventSink.isClosed();
```

Comment:
The method name and @Path value are misleading, considering your return eventSink.isClosed(). How about adding a "!" and reversing the checks in the client?

Reply:
Another stupid mistake and it is fixed now.